internal/impl/aws/input_kinesis.go

```diff
@@ -604,7 +607,11 @@ func (k *kinesisReader) runBalancedShards() {
 			totalShards := len(shardsRes.Shards)
 			unclaimedShards := make(map[string]string, totalShards)
 			for _, s := range shardsRes.Shards {
-				if !isShardFinished(s) {
+				// Update metric for each shard
+				if isShardFinished(s) {
+					metricKVectorShardStatus.With(streamID, *s.ShardId, "finished").Set(1)
```

Comment:
nit: Wondering if we could just do `set(1)` and `set(0)` rather than have `finished` and `open`?
I mean having:
```
	metricKVectorShardStatus := k.mgr.Metrics().GetGaugeVec("kinesis_shard_status", "stream", "shard_id",)
```
Because status is in the name of the metric anyway and then set this to either 0 or 1.

Reply:
not fussed with it, my only nit pick is that it is unclear what 1 or 0 means so we should rename the metric to be either `kinesis_shard_open`/ `kinesis_shard_closed`

Reply:
Well I guess usually 1 means open and 0 means close right? But I'm also not biased towards this. So we can go with the current approach, or what I suggested, or `kinesis_shard_open/ kinesis_shard_closed`. Up to you

Reply:
Ive changed it to `kinesis_shard_open` using 1 and 0